OsmAnd/src/net/osmand/plus/mapcontextmenu/MenuBuilder.java

```diff
@@ -1381,11 +1400,19 @@ private PoiUIFilter getPoiFilterForAmenity(Amenity amenity) {
 	}
 
 	private List<Amenity> getSortedAmenities(PoiUIFilter filter, final LatLon latLon) {
-		QuadRect rect = MapUtils.calculateLatLonBbox(latLon.getLatitude(), latLon.getLongitude(), 250);
+		int radius = 250;
+		QuadRect rect = MapUtils.calculateLatLonBbox(latLon.getLatitude(), latLon.getLongitude(), radius);
 
 		List<Amenity> nearestAmenities = getAmenities(rect, filter);
 		nearestAmenities.remove(amenity);
 
+		boolean isWikiFilter = filter.getFilterId().equals("std_osmwiki");
+
+		for (; nearestAmenities.size() < 10 && !isWikiFilter; radius += 10) {
```

Comment:
- Radius should never increase by constant only by factor 
- there is no check for max radius
- why wikipedia is exceptional?